src/app/_gaurds/auth.guard.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+import { Router, CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot } from '@angular/router';
+import { OAuthService } from 'angular-oauth2-oidc';
+
+@Injectable()
+export class AuthGuard implements CanActivate {
+
+    constructor(
+        private router: Router,
+        private oauthService: OAuthService
+    ) { }
+
+    canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot) {
+        if (!this.oauthService.getAccessToken()) {
+            this.router.navigate(['']);
+            return false;
+        }
+
+        return true;
+    }
+}
```

Comment:
Best practice is to add an additional blank line at the end of the file (Unix file conventions)